lib/puppet/util/posix.rb

```diff
@@ -24,6 +30,35 @@ def groups_of(user)
 
       uniq_groups
     end
+
+    def get_groups_list(user)
```

Comment:
This method may find better place in Etc module (or if you this is better here, make it private)